src/common/snippets/src/pass/collapse_subgraph.cpp

```diff
@@ -167,14 +165,31 @@ auto is_supported_op(const std::shared_ptr<const Node> &n) -> bool {
         return false;
     };
 
+    auto is_supported_reduce_op = [](const std::shared_ptr<const Node> &n) -> bool {
+        if (ov::is_type<const ov::op::v1::ReduceMax>(n) || ov::is_type<const ov::op::v1::ReduceSum>(n)) {
+            const auto& reduce_base = ov::as_type_ptr<const ov::op::util::ArithmeticReductionKeepDims>(n);
+            const auto& axis_constant = ov::as_type_ptr<const ov::op::v0::Constant>(n->get_input_node_shared_ptr(1));
+            if (!reduce_base->get_keep_dims() || !axis_constant || shape_size(axis_constant->get_shape()) != 1)
+                return false;
+
+            const auto rank = n->get_input_partial_shape(0).rank();
+            const auto axis_value = axis_constant->cast_vector<int32_t>(1)[0];
+            const auto normalized_axis = ov::util::normalize_axis(n->get_friendly_name(), axis_value, rank);
+            // Note: Reduction only over the last dimension is currently supported
+            return normalized_axis == rank.get_length() - 1;
```

Comment:
I'd suggest to add check just in case that `rank` is static. We actually have this check in `supported_in_out` but this method is called after `is_support_reduce_op`. (Because before we have the similar bug)

Reply:
Good point, I added the check. Thanks!